static/app/views/settings/account/notifications/notificationSettingsByTypeV2.tsx

```diff
@@ -0,0 +1,338 @@
+          getData: data => {
+            return {
+              type: notificationType,
+              scopeType: 'user',
+              scopeIdentifier: ConfigStore.get('user').id,
+              value: data[notificationType],
+            };
+          },
+        }
+      );
+      fields.push(defaultField);
+    }
+
+    return fields;
+  }
+
+  getProviderFields(): Field[] {
+    const {notificationType} = this.props;
+    const {organizationIntegrations} = this.state;
+    const defaultField = Object.assign({}, NOTIFICATION_SETTING_FIELDS.provider, {
+      choices: organizationIntegrations
+        .map(organizationIntegration => [
+          organizationIntegration.provider.slug,
+          organizationIntegration.provider.name,
+        ])
+        .concat([['email', 'Email']]),
+      getData: data => {
+        return {
+          type: notificationType,
+          scopeType: 'user',
+          scopeIdentifier: ConfigStore.get('user').id,
+          providers: data.provider,
+          value: data[notificationType],
+        };
+      },
+    });
+    const fields: Field[] = [defaultField];
+    return fields;
+  }
+
+  getUnlinkedOrgs = (): OrganizationSummary[] => {
+    const {organizations} = this.props;
+    const {identities, organizationIntegrations} = this.state;
+    const integrationExternalIDsByOrganizationID = Object.fromEntries(
+      organizationIntegrations.map(organizationIntegration => [
+        organizationIntegration.organizationId,
+        organizationIntegration.externalId,
+      ])
+    );
+
+    const identitiesByExternalId = Object.fromEntries(
+      identities.map(identity => [identity?.identityProvider?.externalId, identity])
+    );
+
+    return organizations.filter(organization => {
+      const externalID = integrationExternalIDsByOrganizationID[organization.id];
+      const identity = identitiesByExternalId[externalID];
+      return identity === undefined || identity === null;
+    });
+  };
+
+  handleRemoveNotificationOption = async (id: string) => {
+    await this.api.requestPromise(`/users/me/notification-options/${id}/`, {
+      method: 'DELETE',
+    });
+    this.setState(state => {
+      const newNotificationOptions = state.notificationOptions.filter(
+        option => !(option.id.toString() === id.toString())
+      );
+      return {
+        ...state,
+        notificationOptions: newNotificationOptions,
+      };
+    });
+  };
+
+  handleAddNotificationOption = async (data: Omit<NotificationOptionsObject, 'id'>) => {
+    // TODO: add error handling
+    const notificationOption = await this.api.requestPromise(
+      '/users/me/notification-options/',
+      {
+        method: 'PUT',
+        data,
+      }
+    );
+
+    this.setState(state => {
+      const newNotificationOptions = state.notificationOptions.concat([
+        notificationOption,
+      ]);
+      return {
+        ...state,
+        notificationOptions: newNotificationOptions,
+      };
```

Comment:
maybe i just don't like concat when we have spread
```suggestion
      return {
        ...state,
        notificationOptions: [
        	...state.notificationOptions,
        	notificationOption,
        ],
      };
```